pkg/reconciler/taskrun/taskrun.go

```diff
@@ -299,10 +322,28 @@ func (c *Reconciler) checkPodFailed(ctx context.Context, tr *v1.TaskRun) (bool,
 							}
 						}
 					}
+					// ImagePullBackOff timeout exceeded or not configured
+					image := sidecar.ImageID
+					message := fmt.Sprintf(`the sidecar %q in TaskRun %q failed to pull the image %q. The pod errored with the message: "%s."`, sidecar.Name, tr.Name, image, sidecar.Waiting.Message)
```

Comment:
Optional nit: no need to alloc
```suggestion
					message := fmt.Sprintf(`the sidecar %q in TaskRun %q failed to pull the image %q. The pod errored with the message: "%s."`, sidecar.Name, tr.Name, sidecar.ImageID, sidecar.Waiting.Message)
```